satpy/readers/aapp_l1b.py

```diff
@@ -36,7 +36,22 @@
 from satpy import CHUNK_SIZE
 from satpy.readers.file_handlers import BaseFileHandler
 
-LINE_CHUNK = CHUNK_SIZE ** 2 // 2048
+CHANNEL_DTYPE = float
+
+
+def get_chunks(shape):
```

Comment:
yes, we could make this a generic function, but maybe we should leave the normalize_chunks out of it so that each reader can just call `normalize_chunks` with it's preferred parameters (position of "auto", dtype). So this function would just be actually `get_limit_from_pytroll_chunk_size` or something, that can be removed when we deprecated the environment variable.

`normalize_chunks` does a good size at keeping the user preferences and splitting things evenly, for example if we think of AHI:
```
In [5]: da.core.normalize_chunks((2200, "auto"), shape=(22000, 22000), dtype='float')
Out[5]: 
((2200, 2200, 2200, 2200, 2200, 2200, 2200, 2200, 2200, 2200),
 (5500, 5500, 5500, 5500))
```

So, in summary, I think we should leave `normalize_chunks` to the individual readers, and have them call this function for getting the limit.

Reply:
Some complications/questions with this idea:

1. `CHANNEL_DTYPE` is not an argument to this function, it should be.
2. Couldn't `("auto", 2048)` be an argument to the function too?

If the above is done then I don't see a reason it couldn't stay with the same functionality. This way also the function could be updated with a very descriptive/detailed docstring to help new reader developers understand the decisions that need to be made when choosing chunks for a reader. Additionally, having a central function for controlling things as important as chunking might be nice.

We may need to think about how something like resampling might use this though. For example, the output chunks of resampling based on a target AreaDefinition. That could be handled by this method as well. Or can you think of a reason that it would need a separate `satpy.config` configuration option on top of dask's chunk size config?

Reply:
That's the thing, at the moment PYTROLL_CHUNK_SIZE know nothing of prefered chunks size or dtype, so I think that keeping the same behaviour (just counting the number of pixels) through having this `get_limits` function is good.
After that, I don't see why we shouldn't use `normalize_chunks` directly, that does basically what we need, and the reader dev just has to look at that documentation to understand how to get the chunks sizes they think is best for that specific reader. Or am i missing something?

Reply:
Ah for some reason I read the `if` statement cases backwards and thought the `else:` was when they specified a string. Right now PYTROLL_CHUNK_SIZE is used in a lot of places (even in pyresample I think), so it will be a long time before we can completely deprecated it I think.

I was going to say what if we just completely removed it one reader at a time and had a deprecation warning saying that PYTROLL_CHUNK_SIZE isn't having an effect. But I'm not sure DeprecatingWarning is supposed to be used after the feature is completely removed. Plus that warning could go in this limit function. So I support your idea of moving the limit calculation to a helper function with an added deprecation warning (which don't show up by default). In the future I think we can make it a FutureWarning or even a UserWarning when all uses are removed.

That said, is there any reason to support `PYTROLL_CHUNK_SIZE` being a string in this function? It won't actually work until all use cases of PYTROLL_CHUNK_SIZE are updated or removed. Maybe just a warning pointing people to the dask config setting documentation?

Reply:
Indeed, if we plan on deprecating the use of CHUNK_SIZE entirely, then we don't need to check if it is a string. What I thought when I read this is that we would evolve CHUNK_SIZE to accept the same arguments as `normalize_chunks`, but maybe deprecating this entirely is best.

Reply:
So in summary, I can change this function to just return a limit and put it in `satpy.utils`